src/components/helpers/dom_dnd_hook.ts

```diff
@@ -11,8 +13,77 @@ interface DragAndDropItems extends DragAndDropItemsPartial {
   positionAtStart: Pixel;
 }
 
-export class DOMDndHelper {
+interface DndPartialArgs {
   draggedItemId: UID;
+  initialMousePosition: Pixel;
+  direction: "horizontal" | "vertical";
+  items: DragAndDropItemsPartial[];
+  containerEl: HTMLElement;
+  onChange?: () => void;
+  onCancel?: () => void;
+  onDragEnd?: (itemId: UID, indexAtEnd: Pixel) => void;
+}
+
+type DndArgs = Required<DndPartialArgs>;
+
+interface State {
+  itemsStyle: Record<UID, string>;
+  draggedItemId: UID | undefined;
+  startDnd: (args: DndPartialArgs) => void;
+  cancelDnd: () => void;
+}
+
+export function useDOMDnd() {
```

Comment:
My name kinda sucks, but I don't want a simple `useDnd` name because we already have a helper `startDnd`, and things are confusing. `useDOMDragAndDrop` ? `useElementAndDrop` ? Rename the `startDnd` helepr ? Idk

Reply:
Actually, this hook only handles drag and dropping items from a list. The name could reflect that. Maybe `useDragAndDropListItems` ?